udata/tests/frontend/test_topic_frontend.py

```diff
@@ -3,10 +3,73 @@
 
 from flask import url_for
 
+from udata import search
+from udata.tests import SearchTestMixin, TestCase
 from udata.tests.frontend import FrontTestCase
 from udata.core.dataset.factories import VisibleDatasetFactory
+from udata.core.dataset.search import DatasetSearch
 from udata.core.reuse.factories import VisibleReuseFactory
 from udata.core.topic.factories import TopicFactory
+from udata.core.topic.views import topic_search_for
+
+
+class TopicSearchTest(SearchTestMixin, TestCase):
+    def test_empty_search_no_match(self):
+        '''Should return no result if no data match the tags'''
+        with self.autoindex():
+            VisibleDatasetFactory.create_batch(2, tags=['whatever'])
+        topic = TopicFactory(tags=['no-match'])
+
+        query = topic_search_for(topic, DatasetSearch)
+        result = search.query(query)
+
+        self.assertEqual(len(result), 0)
+
+    def test_empty_search_with_match(self):
+        '''Should only return no data with at least one tag'''
```

Comment:
`Should only return data matching topic's tag`? Otherwise I don't get the logic?